packages/core/src/Dialog/Dialog.styled.tsx

```diff
@@ -0,0 +1,183 @@
+import type { IDialogProps } from '..'
+import { Box, CloseButton, Grid, Text, ThemeProvider } from '..'
+
+import * as Dialog from '@radix-ui/react-dialog'
+import { VisuallyHidden } from '@radix-ui/react-visually-hidden'
+import themeGet from '@styled-system/theme-get'
+import { motion } from 'framer-motion'
+import React from 'react'
+import styled from 'styled-components'
+
+export const DialogSizes = ['sm', 'md', 'lg', 'xl', 'full'] as const
+export type DialogSize = (typeof DialogSizes)[number]
+
+type SizeStyle = {
+  width?: string
+  height?: string
+  borderRadius?: string
+}
+
+const sizeStyles: Record<DialogSize, SizeStyle> = {
+  sm: {
+    width: 'calc(min(400px, 100vw) - 2 * 16px)',
+    borderRadius: '24px',
+  },
+  md: {
+    width: 'calc(min(640px, 100vw) - 2 * 16px)',
+    borderRadius: '24px',
+  },
+  lg: {
+    width: 'calc(min(960px, 100vw) - 2 * 16px)',
+    borderRadius: '24px',
+  },
+  xl: {
+    width: 'calc(min(1280px, 100vw) - 2 * 16px)',
+    height: 'calc(100% - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  full: {
+    width: '100%',
+    height: '100%',
+  },
+} as const
+
+const FloatingCloseButton: typeof CloseButton = styled(CloseButton)`
+  position: absolute;
+  top: ${(props) => -((props.dialogSize === 'full' || props.fullWidth ? 0 : 16) + 10)}px;
+  right: ${(props) => -((props.dialogSize === 'full' || props.fullWidth ? 0 : 16) + 10)}px;
+  margin: ${themeGet('space.3')};
+  padding: ${themeGet('space.2')};
+  background-color: ${themeGet('palette.background.lightest')};
+  box-shadow: ${themeGet('shadows.sm')};
+  &:hover {
+    background-color: ${themeGet('palette.background.lightest')};
+  }
+`
+
+const DialogOverlayWrapper = styled(motion.div)`
+  position: fixed;
+  inset: 0;
+  display: grid;
+  overflow-y: auto;
+  cursor: ${(props: IDialogProps) => (props.scrimDismiss ? 'pointer' : 'default')};
+  place-items: ${(props: IDialogProps) => (props.sheet ? 'end center' : 'center')};
+  background-color: ${(props: IDialogProps) => `rgba(0, 0, 0, ${props.scrimOpacity})`};
+`
+
+const DialogContentWrapper = styled(motion.div)`
+  position: relative;
+  cursor: initial;
+  background-color: ${themeGet('palette.background.lightest')};
+  width: ${(props: IDialogProps) => (props.fullWidth ? '100%' : sizeStyles[props.size]?.width)};
+  height: ${(props: IDialogProps) =>
+    props.sheet ? `calc(${sizeStyles[props.size]?.height} - 24px)` : sizeStyles[props.size]?.height};
+  margin: ${(props: IDialogProps) => (props.sheet ? '16px 0 0 0' : '32px 16px 64px 16px')};
+  border-radius: ${(props: IDialogProps) =>
+    props.sheet ? '24px 24px 0 0' : sizeStyles[props.size]?.borderRadius};
```

Comment:
It might look verbose, but I think we should be getting the individual border-radius values from `theme`.

Reply:
I agree, and I made this change. But then I went behind, and added `borderRadius` as a prop for the Dialog as a whole 😄